File: 30 days of code( c++)/day4.cpp
#include <iostream>
using namespace std;

class Person{
    public:
        int age;
        Person(int initialAge);
        void amIOld();
        void yearPasses();
    };

    Person::Person(int initialAge){
        // Add some more code to run some checks on initialAge
if(initialAge<0)
age=0;
else
age=initialAge;    }

    void Person::amIOld(){
        // Do some computations in here and print out the correct statement to the console 
        if(age==0)
        printf("Age is not valid, setting age to 0.\n");
      if(age<13)
printf("You are young.\n");
else if(age>=13 && age<18)
printf("You are a teenager.\n");
else
printf("You are old.\n");
    }

    void Person::yearPasses(){
        // Increment the age of the person in here
age++;
    }

int main(){
    int t;
	int age;
    cin >> t;
    for(int i=0; i < t; i++) {
    	cin >> age;
        Person p(age);
        p.amIOld();
        for(int j=0; j < 3; j++) {
        	p.yearPasses(); 
        }
        p.amIOld();
      
		cout << '\n';
    }

    return 0;
}

File: 30 days of code( python)/day4.py
class Person:
    # a variable named age for class  
    age = 0

    def __init__(self,initialAge):
        # checking if initialage is negative or not
        if initialAge < 0:
            print("Age is not valid, setting age to 0.")
        else:
            #else assigning value to member variable
            self.age = initialAge

    def amIOld(self):
        # checking value of age using if, elif and else
        if self.age < 13:
            print("You are young.")
        elif self.age >= 13 and self.age < 18 :
            print("You are a teenager.") 
        else:
            print("You are old.")

    def yearPasses(self):
        # Increment the value of age member variable by 1 as the year passes
        self.age = self.age + 1


t = int(input())
for i in range(0, t):
    age = int(input())         
    p = Person(age)  
    p.amIOld()
    for j in range(0, 3):
        p.yearPasses()       
    p.amIOld()
    print("")